docs/contrib-guide.md

```diff
@@ -0,0 +1,158 @@
+# Contribution Guide
+
+1. [Prerequisites](#prerequisites)
+2. [Build](#build)
+3. [Check](#check)
+4. [Testing](#testing)
+5. [Pull-request](#pull-request)
+6. [License](#license)
+
+Thank you for your time and effort to help us improve samba-operator.
+Here are a few steps to get started, with reference to some useful
+resources.
+
+
+## Prerequisites
+
+Development effort takes place using Linux environment and requires at
+minimum:
+
+1. [Go 1.18](https://golang.org/dl/) installed
+2. [GitHub](https://github.com/) account
+3. Development tools: git and make
+4. Testing: [minikube](https://minikube.sigs.k8s.io)
+
+Development collaboration takes place via github's samba-in-kuberbetus
+([SINK](https://github.com/samba-in-kubernetes/)) pages and facilities.
+
+The top-level [Makefile](../Makefile) is the entry point for various
+build commands. Few utilities are required during the build process,
+either for bootstrapping the project, building it or checking the
+validity of the source code itself. If not present in your `$PATH`,
+those tools may be installed locally to the project (under the
+local `.bin` directory) using:
+
+```sh
+  $ make build-tools
+  ...
+  $ ls .bin/
+```
+
+## Build
+
+The following section describes how to build and test samba-operator.
+A more detailed description of this process may be found in the
+[developer-notes](./developer-notes.md) document.
+
+Building the samba-operator is straightforward action with `make build`.
+Upon successful build, the output binary is written to `bin/manager`:
+
+```sh
+  $ make build
+  ...
+  $ ls bin/
+  manager
+```
+
+A developer may override some of the default Makefile settings via
+the `devel.mk` file at the project's root directory:
+
+```sh
+  $ cp devel.mk.sample devel.mk
+  $ vim devel.mk
+```
+
+## Check
+
+Before submitting any pull request, a developer must ensure that
+his code satisfy the following minimum code-style requirements:
+
+1. Go code conform to the project's standards:
+```sh
+  $ make check-format
+  $ make check-golangci-lint
+  $ make check-revive
+  $ make check-gosec
+```
+
+2. YAML files conform to the project's standards:
+```sh
+  $ make check-yaml
+```
+
+3. Unit-tests pass without errors:
+```sh
+  $ make test
+```
+
+
+The above steps may also be executed using single make rule:
+`make check test`. If any of those steps fail they must first be
+fixed before any other developer will review your patches.
```

Comment:
Ah, here it is.  Maybe move this up top to tell people the simpler approach first and then mention how `make check` is a composite of other commands?

Reply:
ACK

Reply:
> Ah, here it is. Maybe move this up top to tell people the simpler approach first and then mention how `make check` is a composite of other commands?

I suggest emphasizing tht the contributor is encouraged to perform those tests/checks on their local changes but that the CI will automatically run them and changes can only be added to the project if all those automated ci checks  pass